docs/metrics/iou.md

```diff
@@ -0,0 +1,170 @@
+# Intersection over Union (IoU)
+
+Intersection over Union (IoU) measures the ratio of the intersection and the union between two instances, ranging from 0 to 1 where 1 indicates a perfect match.  It is one of the metrics that measure the similarity between two instances, often used to compare predictions to ground truths.
+
+As the name suggests, the IoU of two instances ($A$ and $B$) is defined as:
+
+$$IoU(A, B) = \frac {A \cap B} {A \cup B}$$
+
+
+## When Do I Use IoU?
+It is often used to compare two geometries (e.g., bounding boxes, polygons) or segmentation masks in object detection, instance segmentation, or semantic segmentation models. In multi-label classification, IoU, more likely known as the [Jaccard index](https://en.wikipedia.org/wiki/Jaccard_index), is used to compare set of prediction labels for a sample to the corresponding set of ground truth labels. Moreover, there are workflows such as [action detection](https://paperswithcode.com/task/action-detection) and [video moment retrieval](https://paperswithcode.com/task/moment-retrieval) where IoU measures the **temporal** overlap between two time-series snippets.
```

Comment:
> e.g. bounding boxes, polygons

Would be nice to link to the client definitions, e.g. `[bounding boxes][kolena.workflow.annotation.BoundingBox]`

Reply:
> object detection, instance segmentation, or semantic segmentation models

"object detection, instance segmentation, or semantic segmentation workflows" ?